mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/ExtensionTypesDeclarationEnricher.java

```diff
@@ -61,21 +56,39 @@ private void declareSubTypes(ExtensionDeclarer declarer) {
   private void declareDefaultTypes(final ExtensionDeclarer declarer) {
     new IdempotentDeclarationWalker() {
 
-      @Override
-      protected void onParameter(ParameterGroupDeclaration parameterGroup, ParameterDeclaration declaration) {
-        registerType(declarer, declaration.getType());
-      }
-
       @Override
       public void onSource(SourceDeclaration declaration) {
+        registerParametersType(declaration);
         registerType(declarer, declaration);
       }
 
       @Override
       public void onOperation(OperationDeclaration declaration) {
+        registerParametersType(declaration);
         registerType(declarer, declaration);
       }
 
+      @Override
+      protected void onConnectionProvider(ConnectionProviderDeclaration declaration) {
+        registerParametersType(declaration);
+      }
+
+      @Override
+      protected void onConfiguration(ConfigurationDeclaration declaration) {
+        registerParametersType(declaration);
```

Comment:
how is this related to this PR? Is this a new thign?

Reply:
IdempotentDeclarationWalker now walks FunctionDeclarations too, as a side effect, the type of the function parameters would be added to the xml schema if we still do it on the onParameter callback. Because of this, now we register every component parameters individually.